core/include/io.h

```diff
@@ -287,14 +288,13 @@ static inline void io_clrsetbits8(vaddr_t addr, uint8_t clear_mask,
  */
 #define IO_READ32_POLL_TIMEOUT(_addr, _val, _cond, _delay_us, _timeout_us) \
 	({ \
-		uint32_t __timeout = 0; \
+		uint64_t __timeout = timeout_init_us(_timeout_us); \
 		uint32_t __delay = (_delay_us); \
 		\
-		while (__timeout < (_timeout_us)) { \
+		while (!timeout_elapsed(__timeout)) { \
```

Comment:
This adds a dependency on timer counters (`read_cntpct()`/`read_cntvct()`), is it OK? I believe it is but just to make sure.

Reply:
Sole current platforms using the macro are platforms supporting `timeout_xxx()` functions (plat-d06 and plat-stm32mp*) so I guess it's OK but maybe i'm too confident. The alternate easiest way is to force use of a 1us delay in the macro when argument `_delay_us` is 0. What do you think?

Reply:
Or `static_assert(IS_ENABLED(CFG_CORE_HAS_GENERIC_TIMER))`?

Reply:
Are you ok we use `timeout_xxx()` functions in this macro and address the issue later if some platform does not meet this requirement?

Reply:
Yes, absolutely.

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

Reply:
Ok thanks for the feedback.